packages/next/src/server/app-render/make-get-server-inserted-html.tsx

```diff
@@ -69,16 +81,21 @@ export function makeGetServerInsertedHTML({
               return <script key={polyfill.src} {...polyfill} />
             })
         }
-        {renderServerInsertedHTML()}
+        {serverInsertedHTML}
         {errorMetaTags}
-      </>
+      </>,
+      {
+        // Larger chunk because this isn't sent over the network.
+        // Let's set it to 1MB.
+        progressiveChunkSize: 1024 * 1024,
+      }
     )
 
     hasUnflushedPolyfills = false
 
-    // Wait for the stream to be ready.
-    await stream.allReady
-
+    // There's no need to wait for the stream to be ready
+    // e.g. calling `await stream.allReady` because `streamToString` will
+    // wait and decode the stream progressively with better parallelism.
     return streamToString(stream)
```

Comment:
Another low hanging fruit there is that maybe we can just call renderToString in some places where we know there won't be anything that needs to be streamed?

Reply:
Yes that would require a larger change but this is definitely worth it (blocked by our edge React server not having renderToString exported). It will make this insertion function sync, and all the underlying stream methods sync too.